python/tank/util/system_settings.py

```diff
@@ -15,12 +15,23 @@
 
 import urllib
 
+from .singleton import Singleton
+from .. import LogManager
 
-class SystemSettings(object):
+logger = LogManager.get_logger(__name__)
+
+
+class SystemSettings(Singleton):
     """
     Handles loading the system settings.
     """
 
+    def _init_singleton(self):
+        """
+        Singleton initialization.
+        """
+        logger.debug("OS http proxy: %s", self.http_proxy)
```

Comment:
Any worries about logging passwords here?

Reply:
The urllib library doesn't support proxies by default, but it's probably not safe to assume that the getproxies method doesn't  return the value straight from the OS, which might be password protected. 

Maybe we should refactor [this](https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/util/user_settings.py#L256-L270)